doc/15-troubleshooting.md

```diff
@@ -230,6 +232,97 @@ icinga2 7764 nagios  mem    REG                8,5   744776 2631636 /usr/lib/x86
 #
 ```
 
+### Optimise apply rules and group assign conditions
+
+#### Remove actually unused apply rules
+
+If `icinga2 daemon -C` warns you like shown below and "it's okay" for you
+as everything behaves as desired, consider removing
+or commenting out the respective apply rule.
+
+```
+[...] warning/ApplyRule: Apply rule '...' (...) for type '...' does not match anywhere!
+```
+
+At least consider changing the `assign where` filter from, say
+`assign where my_complex_filter()`, to `assign where false && my_complex_filter()`.
+`&&` will evaluate `false` first and in an instant.
+If it's not true, `&&` will "return" `false` and skip `my_complex_filter()`.
+
+Same for `assign where` conditions in groups. In this case removing just
+the `assign where` line(s) is enough if you'd like to keep the group itself.
+
+#### Create single objects not via apply rules
+
+If possible, replace constructs like the immediately following with the below one.
+
+```
+apply Service "syspatch" {
+	// ...
+	assign where host.name == "firewall" || host.name == "ssh-jump-box"
+}
+```
+
+```
+object Service "syspatch" {
+	// ...
+	host_name = "firewall"
+}
+
+object Service "syspatch" {
+	// ...
+	host_name = "ssh-jump-box"
+}
+```
+
+This way Icinga won't have to evaluate
+`host.name == "firewall" || host.name == "ssh-jump-box"`
+for all of your hosts.
```

Comment:
I think having a single host name in here would suffice. Strictly speaking, that example doesn't even create a single object.